src/components/ProfilePopover/index.tsx

```diff
@@ -1,78 +1,78 @@
 import { useEffect, useState } from "react";
+import Image from "next/image";
 import { UserResource } from "@clerk/types";
 import { Popover, Transition } from "@headlessui/react";
 import { GoCheck, GoPencil, GoX } from "react-icons/go";
 
 import { useAuth } from "@/hooks/auth";
-import { UserRole } from "@/types";
 
-interface Props {
-  firstName: string | null | undefined;
-  lastName: string | null | undefined;
-  role: UserRole;
-  desc: string | null | undefined;
+async function updateUserMetadata(user: UserResource, description: string) {
+  await user.update({ unsafeMetadata: { description } });
 }
 
-const updateUserMetadata = async (
-  user: UserResource | null | undefined,
-  description: string
-) => {
-  await user?.update({
-    unsafeMetadata: {
-      description: description
-    }
-  });
-};
+export default function ProfilePopover() {
+  const { user, role } = useAuth();
 
-export default function ProfilePopover({
-  firstName,
-  lastName,
-  role,
-  desc
-}: Props) {
   const [isEdit, setIsEdit] = useState(false);
-  const [description, setDescription] = useState(desc);
-  const [initialDescription, setInitialDescription] = useState(desc);
-  const { user } = useAuth();
-
-  // if (firstName === null || firstName === undefined) return null;
+  const [description, setDescription] = useState(
+    String(user?.unsafeMetadata.description) || ""
+  );
+  const [initialDescription, setInitialDescription] = useState(description);
 
   useEffect(() => {
-    setDescription(desc);
-  }, [desc]);
+    setDescription(String(user?.unsafeMetadata.description) || "");
+  }, [user?.unsafeMetadata.description]);
```

Comment:
Can you have possibly undefine variable in the useEffect dependency array?

Reply:
Yeah you can include a potentially undefined variable in the dependency array. It's actually quite common to do so, especially when dealing with asynchronous data fetching or props that might change over time.

When you include a variable that can be undefined in a useEffect dependency array, the effect will run whenever the variable changes, including when it becomes undefined or changes from undefined to a defined value. This behavior is often useful for responding to changes in data that might be loaded asynchronously.